staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go

```diff
@@ -537,18 +537,35 @@ func (b *builder) getOpenAPIConfig(v2 bool) *common.Config {
 			buildDefinitions.Do(generateBuildDefinitionsFunc)
 			return namer.GetDefinitionName(name)
 		},
-		GetDefinitions: func(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
-			def := utilopenapi.GetOpenAPIDefinitionsWithoutDisabledFeatures(generatedopenapi.GetOpenAPIDefinitions)(ref)
-			def[fmt.Sprintf("%s/%s.%s", b.group, b.version, b.kind)] = common.OpenAPIDefinition{
-				Schema:       *b.schema,
-				Dependencies: []string{objectMetaType},
-			}
-			def[fmt.Sprintf("%s/%s.%s", b.group, b.version, b.listKind)] = common.OpenAPIDefinition{
-				Schema: *b.listSchema,
-			}
-			return def
-		},
+		// GetDefinitions: func(ref common.ReferenceCallback) map[string]common.OpenAPIDefinition {
+		// 	def := utilopenapi.GetOpenAPIDefinitionsWithoutDisabledFeatures(generatedopenapi.GetOpenAPIDefinitions)(ref)
+		// 	def[fmt.Sprintf("%s/%s.%s", b.group, b.version, b.kind)] = common.OpenAPIDefinition{
+		// 		Schema:       *b.schema,
+		// 		Dependencies: []string{objectMetaType},
+		// 	}
+		// 	def[fmt.Sprintf("%s/%s.%s", b.group, b.version, b.listKind)] = common.OpenAPIDefinition{
+		// 		Schema: *b.listSchema,
+		// 	}
+		// 	return def
+		// },
+	}
+	buildDefinitions.Do(generateBuildDefinitionsFunc)
+	if v2 {
+		c.Definitions = definitions
+	} else {
+		c.Definitions = definitionsV3
+	}
+	def := make(map[string]common.OpenAPIDefinition)
+	def[fmt.Sprintf("%s/%s.%s", b.group, b.version, b.kind)] = common.OpenAPIDefinition{
+		Schema:       *b.schema,
+		Dependencies: []string{objectMetaType},
 	}
+	def[fmt.Sprintf("%s/%s.%s", b.group, b.version, b.listKind)] = common.OpenAPIDefinition{
+		Schema: *b.listSchema,
+	}
+	c.AdditionalDefinitions = def
```

Comment:
What is the difference between `definitions` and `additionalDefinitions`?

Reply:
Might be more clear after a rename but
definitions = built in static definitions that can be fixed and reused
additionalDefinitions = CRD definitions
kube-openapi should treat them as just two maps that should be union'd together when searching